### PR TITLE
Compile for torchtitan to match vllm compile

### PR DESCRIPTION
**Pull request by PaulZhang12**

### CODE REVIEW

**Comment by Lucaskabela** [comment]
Here is a more updated version working on my branch - will follow up with cleaning, minimizing, and investigating the issues present (recompile, differing logits, speedup)

https://github.com/Lucaskabela/torchtitan/pull/1
